group_vars/matrix_servers

```diff
@@ -1538,6 +1538,26 @@ matrix_jitsi_etherpad_base: "{{ matrix_etherpad_base_url if matrix_etherpad_enab
 # /matrix-jitsi
 #
 ######################################################################
+######################################################################
+#
+# matrix-ldap-registration-proxy
+#
+######################################################################
+
+# This is only for users with a specific LDAP setup
+matrix_ldap_registration_proxy_enabled: false
+
+# Use the LDAP values specified for the synapse role to setup LDAP proxy
+matrix_ldap_registration_proxy_ldap_uri: "{{ matrix_synapse_ext_password_provider_ldap_uri }}"
+matrix_ldap_registration_proxy_ldap_base_dn: "{{ matrix_synapse_ext_password_provider_ldap_base }}"
+matrix_ldap_registration_proxy_ldap_user: "{{ matrix_synapse_ext_password_provider_ldap_bind_dn }}"
+matrix_ldap_registration_proxy_ldap_password: "{{ matrix_synapse_ext_password_provider_ldap_bind_password }}"
```

Comment:
Defining these settings in terms of `matrix_synapse_ext_password_provider_ldap_` is convenient if you're switching from the LDAP password provider for Synapse.. However it makes me think that `ldap-registration-proxy` **must** be used together with the LDAP password provider for Synapse. Is that true?

If it can be used on its own, you'd better separate these (by removing all these variable re-definitions here).

If these 2 components really need to be used together, you'd better:

- update [the upstream repository](https://gitlab.com/activism.international/matrix_ldap_registration_proxy), as it's not giving any such indication
- add documentation in `roles/matrix-ldap-registration-proxy/defaults/main.yml`, etc.

----

Speaking of which, adding a `docs/` page about this component and how to install/use it within the context of the playbook would be beneficial.

Does one simply need to set `matrix_ldap_registration_proxy_enabled` to `true` and run the playbook (`--tags=setup-all,start`)? Or does one need to configure the LDAP password provider for Synapse first?

Any other things that need to be done?

Reply:
I don't know. If it needs to be used together we could indicate that in the docs, else I would move this block to group vars and activate it when the LDAP password provider for Synapse is used.
@TheOneWithTheBraid has to answer this, I don't have much knowledge of LDAP setups

Reply:
They are *technically* completely independent from each other even though I's state the ldap_registration_proxy does not make any sense without the LDAP password provider in most cases.

But of course, theoretically, one could use the ldap_registration_proxy in order to register users while any other unknown implementation handles the channel back from LDAP to synapse or maybe any other homeserver implementation.

Reply:
I'd anyway appreciate naming the ldap_registration_proxy on the docs' LDAP page along with the Synapse ext password LDAP and ma1dn LDAP.